compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs

```diff
@@ -640,7 +640,8 @@ fn codegen_regular_intrinsic_call<'tcx>(
         sym::assert_inhabited | sym::assert_zero_valid | sym::assert_mem_uninitialized_valid => {
             intrinsic_args!(fx, args => (); intrinsic);
 
-            let layout = fx.layout_of(substs.type_at(0));
+            let ty = substs.type_at(0);
+            let layout = fx.layout_of(ty);
```

Comment:
It's a bit of a bummer that we now always do 2 layout queries here, but it's probably not too bad.